### PR TITLE
doc: app_development: add header tags

### PR DESCRIPTION
**Pull request by greg-fer**
Added several header tags on Application Development. Edited header hierarchy under Create Application.

### CODE REVIEW

File: doc/develop/application/index.rst [hunk 1]
```diff
@@ -190,9 +190,15 @@ following example, ``app`` is a Zephyr freestanding application:
         └── src/
             └── main.c
 
+.. _zephyr-creating-app:
+
 Creating an Application
 ***********************
 
+In Zephyr, you can either use a reference workspace application or create your application by hand from scratch.
```
Comment (kartben) [inline doc/develop/application/index.rst:198]: "by hand from scratch" sounds redundant?

```suggestion
In Zephyr, you can either use a reference workspace application or create your application by hand.
```

File: doc/develop/application/index.rst [hunk 2]
```diff
@@ -190,9 +190,15 @@ following example, ``app`` is a Zephyr freestanding application:
...
+
+.. _zephyr-creating-app-from-example:
+
 example-application
```
Comment (kartben) [inline doc/develop/application/index.rst:202]: I would rename this to make it clearer this is the first of the two options enumerated just before

```suggestion
Using a Reference Workspace Application (example-application)
```

**Review by kartben — CHANGES_REQUESTED** [review]

File: doc/develop/application/index.rst (hunk 2 above)
Comment (greg-fer) [reply doc/develop/application/index.rst:202]: Even shorter without the "(example-application)" should be fine.

Comment (kartben) [reply doc/develop/application/index.rst]: True :)

**Review by kartben — APPROVED** [review]
Thanks!

**Review by nordicjm — APPROVED** [review]

**Pull request merged**